src/jdk.incubator.code/share/classes/jdk/incubator/code/op/CoreOp.java

```diff
@@ -1278,59 +1278,28 @@ public static ConstantOp create(ExternalizedOp def) {
         }
 
         static Object processConstantValue(TypeElement t, Object value) {
-            if (t.equals(JavaType.BOOLEAN)) {
-                if (value instanceof String s) {
-                    return Boolean.valueOf(s);
-                } else if (value instanceof Boolean) {
-                    return value;
-                }
-            } else if (t.equals(JavaType.BYTE)) {
-                if (value instanceof String s) {
-                    return Byte.valueOf(s);
-                } else if (value instanceof Number n) {
-                    return n.byteValue();
-                }
-            } else if (t.equals(JavaType.SHORT)) {
-                if (value instanceof String s) {
-                    return Short.valueOf(s);
-                } else if (value instanceof Number n) {
-                    return n.shortValue();
-                }
-            } else if (t.equals(JavaType.CHAR)) {
-                if (value instanceof String s) {
-                    return s.charAt(0);
-                } else if (value instanceof Character) {
-                    return value;
-                }
-            } else if (t.equals(JavaType.INT)) {
-                if (value instanceof String s) {
-                    return Integer.valueOf(s);
-                } else if (value instanceof Number n) {
-                    return n.intValue();
-                }
-            } else if (t.equals(JavaType.LONG)) {
-                if (value instanceof String s) {
-                    return Long.valueOf(s);
-                } else if (value instanceof Number n) {
-                    return n.longValue();
-                }
-            } else if (t.equals(JavaType.FLOAT)) {
-                if (value instanceof String s) {
-                    return Float.valueOf(s);
-                } else if (value instanceof Number n) {
-                    return n.floatValue();
-                }
-            } else if (t.equals(JavaType.DOUBLE)) {
-                if (value instanceof String s) {
-                    return Double.valueOf(s);
-                } else if (value instanceof Number n) {
-                    return n.doubleValue();
-                }
+            if (t.equals(JavaType.BOOLEAN) && value instanceof Boolean) {
```

Comment:
Nice. Can the IDE refactor the body of this method to a switch expression?

Reply:
We can't really -- because the switch is on `JavaType` constant (and we don't support constant patterns -- at least yet :-) ). So, at least "as written", we can't easily turn this into a switch.